crates/ra_ide/src/completion/complete_record_literal.rs

```diff
@@ -11,8 +11,24 @@ pub(super) fn complete_record_literal(acc: &mut Completions, ctx: &CompletionCon
         _ => return,
     };
 
+    let already_present_names: Vec<String> = ctx
```

Comment:
And this needs to be a SmallStr then

bors d+

Reply:
Is there any way to get the `SmolStr` out of the `Name` (I see that there's one inside `Repr`, but no method in the public API for that, apparently).

Reply:
Nope, by design. Ideally, I think that this should probably work from hir, using the same data as the missing struct fields diagnostics.

Reply:
The assist compares `Name`'s and uses `ra_hir_def::Expr::RecordLit` for that, but this is a private thing for `ra_ide`, as I see.
It would be really simple to import the `AsName` trait to perform the `NameRef -> Name` conversion, but that is also private for `ra_ide` and I remember you stating that it is a concious decision.

So it appears to me, that the only way to solve this with the current API is to allocate some strings after all: keep the `SmolStr` here, as proposed and do the `&SmolStr::from(field.name(ctx.db).to_string())` when checking for `contains`.

Is there any better way in your opinion?